lib/kube/proxy/sess.go

```diff
@@ -165,34 +165,54 @@ func (p *kubeProxyClientStreams) sendStatus(err error) error {
 }
 
 func (p *kubeProxyClientStreams) Close() error {
-	close(p.close)
 	return trace.Wrap(p.proxy.Close())
 }
 
 // multiResizeQueue is a merged queue of multiple terminal size queues.
 type multiResizeQueue struct {
-	queues   map[string]<-chan *remotecommand.TerminalSize
-	cases    []reflect.SelectCase
-	callback func(*remotecommand.TerminalSize)
-	mutex    sync.Mutex
+	queues       map[string]<-chan *remotecommand.TerminalSize
+	cases        []reflect.SelectCase
+	callback     func(*remotecommand.TerminalSize)
+	mutex        sync.Mutex
+	parentCtx    context.Context
+	reloadCtx    context.Context
+	reloadCancel context.CancelFunc
 }
 
-func newMultiResizeQueue() *multiResizeQueue {
+func newMultiResizeQueue(parentCtx context.Context) *multiResizeQueue {
+	ctx, cancel := context.WithCancel(parentCtx)
 	return &multiResizeQueue{
-		queues: make(map[string]<-chan *remotecommand.TerminalSize),
+		queues:       make(map[string]<-chan *remotecommand.TerminalSize),
+		parentCtx:    parentCtx,
+		reloadCtx:    ctx,
+		reloadCancel: cancel,
 	}
 }
 
 func (r *multiResizeQueue) rebuild() {
-	r.cases = nil
+	oldCancel := r.reloadCancel
+	defer oldCancel()
+
+	r.reloadCtx, r.reloadCancel = context.WithCancel(r.parentCtx)
+	r.cases = make([]reflect.SelectCase, len(r.queues)+1)
+	r.cases[0] = reflect.SelectCase{
+		Dir:  reflect.SelectRecv,
+		Chan: reflect.ValueOf(r.reloadCtx.Done()),
+	}
+	i := 1
 	for _, queue := range r.queues {
-		r.cases = append(r.cases, reflect.SelectCase{
+		r.cases[i] = reflect.SelectCase{
```

Comment:
```suggestion
		r.cases[i+1] = reflect.SelectCase{
```
Then you could drop the `i` variable that you create before the loop, change index variable in the loop from `_` to `i` and then remove the increment operation at the bottom. 

That way we have less code to worry about 😄

Reply:
`r.queues` is a map whose key is a string.

Reply:
Ahh, sorry, my bad!